### DISCUSSION TITLE
Error building service com.polarion.alm.richpages.richPageWidgetContribution

### DISCUSSION THREAD

**Discussion by puravpanchal** [SchweizerischeBundesbahnen/ch.sbb.polarion.extension.pdf-exporter#252] [Q&A] [answered]
Hi, 

I have installed the extension along with the WeasyPrint and have added the extension code in the Project Administration - Document Sidebar Properties editor. 

However, when I open the project it gives the following error:

Time: 2024-10-09 11:54

Browser: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/129.0.0.0 Safari/537.36

Client Permutation: DF75BE4107122C81DD74CDCBC14E8DB7

Product Name: Polarion ALM

Product Version: 22 R1 (Build: 3.22.1.20220419-1528-22_R1-be3adceb)

User's License: ALM

Error message: Unable to construct service com.polarion.alm.richpages.richPageWidgetContribution: Error building service com.polarion.alm.richpages.richPageWidgetContribution: Failure invoking constructor for class com.polarion.alm.server.rp.RichPageWidgetContributionImpl: Unable to construct configuration com.polarion.alm.richpages.widgets: Unable to convert type 'ch.sbb.polarion.extension.pdf_exporter.widgets.ExportToPdfButton' to a Java class, either as is, or in package ch.sbb.polarion.extension.pdf_exporter.

Details: Error ID: 1b7a9952-3b25-4d93-81b1-fc1c1988a9a0
See system log for more information

I have attached the polarion log:
[polarionlog.txt](https://github.com/user-attachments/files/17306949/polarionlog.txt)

Any help is appreciated!!

Thanks
Purav

**Comment by grigoriev** [comment, marked as answer]
Hi @puravpanchal,

the PDF-Exporter extension supports only 2 latest version of Polarion (right now 2310 and 2404). 
Polarion 22R1 uses java 11 that's why you get this error message.

So there is no easy solution: you can update Polarion or compile this extension against Polarion 22R1/Java 11.
